inc/Engine/Media/Fonts/Controller/Filesystem.php

```diff
@@ -0,0 +1,217 @@
+<?php
+
+declare(strict_types=1);
+
+namespace WP_Rocket\Engine\Media\Fonts\Controller;
+
+class Filesystem {
+	/**
+	 * WP Filesystem instance
+	 *
+	 * @var WP_Filesystem_Direct
+	 */
+	private $filesystem;
+
+	/**
+	 * Path to the fonts storage
+	 *
+	 * @var string
+	 */
+	private $path;
+
+	/**
+	 * Version of the fonts.
+	 *
+	 * @var int
+	 */
+	private $version;
+
+	/**
+	 * Instantiate the class
+	 *
+	 * @param string               $base_path Base path to the fonts storage.
+	 * @param WP_Filesystem_Direct $filesystem WP Filesystem instance.
+	 */
+	public function __construct( $base_path, $filesystem = null ) {
+		$this->filesystem = is_null( $filesystem ) ? rocket_direct_filesystem() : $filesystem;
+		$this->path       = $base_path . get_current_blog_id() . '/';
+	}
+
+	/**
+	 * Hash the url
+	 *
+	 * @param string $url Url of the page to hash.
+	 *
+	 * @return string
+	 */
+	private function hash_url( string $url ): string {
+		return md5( $url );
+	}
+
+	/**
+	 * Write font css to path
+	 *
+	 * @param string $font_url The font url to save locally.
+	 * @param string $provider The url of the page.
+	 *
+	 * @return bool
+	 */
+	public function write_font_css( string $font_url, string $provider ): bool {
+		global $wp;
+		$url = untrailingslashit( home_url( add_query_arg( [], $wp->request ) ) );
+
+		$font_provider_path = $this->get_font_provider_path( $provider );
+		$hash_url           = $this->hash_url( $url );
```

Comment:
the hash should be based on the Google Fonts URL, not the current page URL

Reply:
I used the page URL so the frontend can use it to get the fonts and rewrite.